src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -1108,5 +1120,198 @@ public void PopulateCommand(SqlCommand sqlCommand, short resourceTypeId, long st
                 _up.AddParameter(sqlCommand.Parameters, up);
             }
         }
+
+        private class ResourceSearchParamStats
+        {
+            private ISqlRetryService _sqlRetryService;
+            private readonly ILogger<SqlServerSearchService> _logger;
+            private readonly ISqlServerFhirModel _model;
+            private ConcurrentDictionary<(string TableName, string ColumnName, short ResourceTypeId, short SearchParamId), bool> _stats;
```

Comment:
As Init() is only called once, _stats can also be marked as readonly, if its initialization is moved out of Init().

Reply:
Done.